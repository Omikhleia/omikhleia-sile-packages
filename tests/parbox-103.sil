\begin[papersize=a6]{document}
\neverindent
\nofolios
\script[src=packages/parbox]
\script[src=packages/rules]
\script{
SILE.registerCommand("test:fancymargins", function (options, content)
  SILE.settings.set("document.lskip", SILE.length("1em"))
  SILE.settings.set("document.rskip", SILE.nodefactory.hfillglue())
  SILE.settings.set("document.parindent", SILE.length("-1em"))
  SILE.process(content)
end)
}%
%
% End-to-end complex scenarios that push the alignment subtleties
% to the battle ground.

% Just an hrule as visual guideline
0.
\hrule[width=50%fw, height=0.1pt]

% Minimization width re-computation check:
% Parbox with a long line-broken text and some negative spaces.
% 1: with a negative width explicitly introduced via a kern (cancelling a quad)
% 2: negative parindent, ragged right
% 3: negative parindent, justified (regular line-breaking)
1.
\parbox[valign=top, strut=character, width=50%fw, border=0.5pt,
  minimize=true]{\quad{}\kern[width=-1em]{}Lorem ipsum.}

2.
\parbox[strut=character, width=50%fw, border=0.5pt,
  minimize=true]{%
\test:fancymargins{Lorem ipsum dolor sit amet consetetur sadipscing}}

3.
\parbox[strut=character, width=50%fw, border=0.5pt,
  minimize=true]{\test:fancymargins{\set[parameter=document.rskip, value=0]Lorem ipsum dolor sit amet consetetur sadipscing}}

% Verticcal strech/shrink cancellation
% (medskip is variable, so if cancellation is not perfect, the box layout and the
% rendered lines get may misaligned vertically).
4.
\parbox[width=50%fw, strut=character, border=0.5pt]{Lorem ipsum dolor sit amet consetetur sadipscing

\center{centered}

\raggedleft{I am ragged\break left}

\medskip

A paragraph after a skip.
}

\smallskip

% Proper vertical baseline aligment check when the parbox content is bigger
% that the strut (due to font change, images, etc. = emulated here with hrules).
6.
\parbox[valign=top, strut=character, width=25%fw, border=0.5pt]{Lorem
  ipsum \hrule[height=2em, width=2pt]\hrule[depth=1em, width=2pt]\par
  Sit \hrule[height=1em, width=2pt]\hrule[depth=1.5em, width=2pt]}%
...\parbox[valign=bottom, strut=character, width=25%fw, border=0.5pt]{Lorem
  ipsum \hrule[height=2em, width=2pt]\hrule[depth=1em, width=2pt]\par
  Sit \hrule[height=1em, width=2pt]\hrule[depth=1.5em, width=2pt]}%
...\parbox[valign=middle, strut=character, width=25%fw, border=0.5pt]{Lorem
  ipsum \hrule[height=2em, width=2pt]\hrule[depth=1em, width=2pt]\par
  Sit \hrule[height=1em, width=2pt]\hrule[depth=1.5em, width=2pt]}

\end{document}