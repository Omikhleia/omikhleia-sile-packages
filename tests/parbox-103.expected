Set paper size 	297.6377985	419.5275636
Begin page
Mx 	14.8819
My 	27.0067
Set font 	Gentium Plus;10;400;;normal;;LTR
T	19 w=4.6924	(0)
Mx 	19.5743
T	17 w=2.2900	(.)
Draw line	24.5206	26.9067	133.9370	0.1000
Mx 	14.8819
My 	39.0067
T	20 w=4.6924	(1)
Mx 	19.5743
T	17 w=2.2900	(.)
Draw line	24.5291	30.8524	56.5342	0.5000
Draw line	24.5291	41.4481	56.5342	0.5000
Draw line	24.5291	30.8524	0.5000	11.0957
Draw line	81.0633	30.8524	0.5000	11.0957
Mx 	24.5291
T	47 82 85 72 80 w=26.4014	(Lorem)
Mx 	53.5730
T	76 83 86 88 80 w=25.2002	(ipsum)
Mx 	78.7732
T	17 w=2.2900	(.)
Mx 	14.8819
My 	51.0067
T	21 w=4.6924	(2)
Mx 	19.5743
T	17 w=2.2900	(.)
Draw line	24.5227	42.8524	114.7988	0.5000
Draw line	24.5227	65.4481	114.7988	0.5000
Draw line	24.5227	42.8524	0.5000	23.0957
Draw line	139.3215	42.8524	0.5000	23.0957
Mx 	24.5227
T	47 82 85 72 80 w=26.4014	(Lorem)
Mx 	53.5666
T	76 83 86 88 80 w=25.2002	(ipsum)
Mx 	81.4094
T	71 82 79 82 85 w=21.9287	(dolor)
Mx 	105.9807
T	86 76 87 w=10.0146	(sit)
Mx 	118.6379
T	68 80 72 87 w=20.6836	(amet)
Mx 	34.5227
My 	63.0067
T	70 82 81 86 72 87 72 87 88 85 w=44.1602	(consetetur)
Mx 	81.3254
T	86 68 71 76 83 86 70 76 81 74 w=42.9980	(sadipscing)
Mx 	14.8819
My 	74.6024
T	22 w=4.6924	(3)
Mx 	19.5743
T	17 w=2.2900	(.)
Draw line	24.5206	66.4481	133.9370	0.5000
Draw line	24.5206	89.0438	133.9370	0.5000
Draw line	24.5206	66.4481	0.5000	23.0957
Draw line	158.4576	66.4481	0.5000	23.0957
Mx 	24.5206
T	47 82 85 72 80 w=26.4014	(Lorem)
Mx 	53.2064
T	76 83 86 88 80 w=25.2002	(ipsum)
Mx 	80.6911
T	71 82 79 82 85 w=21.9287	(dolor)
Mx 	104.9043
T	86 76 87 w=10.0146	(sit)
Mx 	117.2034
T	68 80 72 87 w=20.6836	(amet)
Mx 	140.1715
T	70 82 81 w=14.9170	(con)
Mx 	155.0884
T	16 w=3.3691	(-)
Mx 	34.5206
My 	86.6024
T	86 72 87 72 w=16.5430	(sete)
Mx 	51.0635
T	87 88 85 w=12.7002	(tur)
Mx 	66.4117
T	86 68 71 76 83 86 70 76 81 74 w=42.9980	(sadipscing)
Mx 	14.8819
My 	98.1981
T	23 w=4.6924	(4)
Mx 	19.5743
T	17 w=2.2900	(.)
Draw line	24.5206	90.0438	133.9370	0.5000
Draw line	24.5206	166.6395	133.9370	0.5000
Draw line	24.5206	90.0438	0.5000	77.0957
Draw line	158.4576	90.0438	0.5000	77.0957
Mx 	24.5206
T	47 82 85 72 80 w=26.4014	(Lorem)
Mx 	53.2064
T	76 83 86 88 80 w=25.2002	(ipsum)
Mx 	80.6911
T	71 82 79 82 85 w=21.9287	(dolor)
Mx 	104.9043
T	86 76 87 w=10.0146	(sit)
Mx 	117.2034
T	68 80 72 87 w=20.6836	(amet)
Mx 	140.1715
T	70 82 81 w=14.9170	(con)
Mx 	155.0884
T	16 w=3.3691	(-)
Mx 	24.5206
My 	110.1981
T	86 72 87 72 w=16.5430	(sete)
Mx 	41.0635
T	87 88 85 w=12.7002	(tur)
Mx 	56.4128
T	86 68 71 76 83 86 70 76 81 74 w=42.9980	(sadipscing)
Mx 	73.3153
My 	122.1981
T	70 72 81 87 72 85 72 71 w=36.3477	(centered)
Mx 	109.4468
My 	134.1981
T	44 w=2.9785	(I)
Mx 	115.0679
T	68 80 w=12.6221	(am)
Mx 	130.3326
T	85 68 74 74 72 71 w=28.1250	(ragged)
Mx 	144.5757
My 	146.1981
T	79 72 73 87 w=13.8818	(left)
Mx 	24.5206
My 	164.1981
T	36 w=5.9521	(A)
Mx 	33.1188
T	83 68 85 68 74 85 68 83 75 w=42.6807	(paragraph)
Mx 	78.4455
T	68 73 87 72 85 w=19.7217	(after)
Mx 	100.8132
T	68 w=4.5898	(a)
Mx 	108.0490
T	86 78 76 83 w=16.7480	(skip)
Mx 	124.7971
T	17 w=2.2900	(.)
Mx 	14.8819
My 	211.6395
T	25 w=4.6924	(6)
Mx 	19.5743
T	17 w=2.2900	(.)
Draw line	24.5117	191.6395	66.9685	0.5000
Draw line	24.5117	247.6395	66.9685	0.5000
Draw line	24.5117	191.6395	0.5000	56.5000
Draw line	91.4802	191.6395	0.5000	56.5000
Mx 	24.5117
T	47 82 85 72 80 w=26.4014	(Lorem)
Mx 	53.5563
T	76 83 86 88 80 w=25.2002	(ipsum)
Draw line	81.3997	191.6395	2.0000	20.0000
Draw line	83.3997	211.6395	2.0000	10.0000
Mx 	24.5117
My 	232.6395
T	54 76 87 w=10.9131	(Sit)
Draw line	38.0728	222.6395	2.0000	10.0000
Draw line	40.0728	232.6395	2.0000	15.0000
Mx 	91.4802
My 	211.6395
T	17 w=2.2900	(.)
Mx 	93.7702
T	17 w=2.2900	(.)
Mx 	96.0603
T	17 w=2.2900	(.)
Draw line	98.3503	170.6395	66.9685	0.5000
Draw line	98.3503	226.6395	66.9685	0.5000
Draw line	98.3503	170.6395	0.5000	56.5000
Draw line	165.3188	170.6395	0.5000	56.5000
Mx 	98.3503
My 	190.6395
T	47 82 85 72 80 w=26.4014	(Lorem)
Mx 	127.3949
T	76 83 86 88 80 w=25.2002	(ipsum)
Draw line	155.2384	170.6395	2.0000	20.0000
Draw line	157.2384	190.6395	2.0000	10.0000
Mx 	98.3503
My 	211.6395
T	54 76 87 w=10.9131	(Sit)
Draw line	111.9114	201.6395	2.0000	10.0000
Draw line	113.9114	211.6395	2.0000	15.0000
Mx 	165.3188
T	17 w=2.2900	(.)
Mx 	167.6089
T	17 w=2.2900	(.)
Mx 	169.8989
T	17 w=2.2900	(.)
Draw line	172.1889	183.6395	66.9685	0.5000
Draw line	172.1889	239.6395	66.9685	0.5000
Draw line	172.1889	183.6395	0.5000	56.5000
Draw line	239.1574	183.6395	0.5000	56.5000
Mx 	172.1889
My 	203.6395
T	47 82 85 72 80 w=26.4014	(Lorem)
Mx 	201.2335
T	76 83 86 88 80 w=25.2002	(ipsum)
Draw line	229.0770	183.6395	2.0000	20.0000
Draw line	231.0770	203.6395	2.0000	10.0000
Mx 	172.1889
My 	224.6395
T	54 76 87 w=10.9131	(Sit)
Draw line	185.7500	214.6395	2.0000	10.0000
Draw line	187.7500	224.6395	2.0000	15.0000
End page
Finish
